### DISCUSSION TITLE
Syncing codespaces?

### DISCUSSION THREAD

**Discussion by allenmoench** [earthlab-education/Earth-Analytics-AY24#353] [Q&A]
Hello! After I completed the Climate Coding Challenge, I couldn't find options to commit or sync changes. Does anyone know how to do that? Thanks!

**Comment by nquarder** [comment]
Hi @allenmoench - good question and sorry for not having a demo video of this up for folks yet - we are working on it! And if you watch the recording in Canvas from my discussion section last week (Weds 1-2:30pm MST) you'll see it demonstrated towards the end of our meeting. 

I'm including screenshots from my codespace that will help hopefully until we get the video up. 

You will notice from the menu on the left side of the screen the little icon with the blue dot and a number 1 in the center. Select this icon (Source Control). From there you will be prompted to write a message that summarizes the work you're completing (_finishing climate challenge_ or something short and descriptive). Then hit the green (or blue depending on your settings) button that says **Commit**. 

<img width="723" alt="Screenshot 2024-09-17 at 9 31 06 AM" src="https://github.com/user-attachments/assets/573e2eaf-317a-40a6-aec8-21d8ebd2cc58">

Select **Yes** when asked about staged changes.

<img width="618" alt="Screenshot 2024-09-17 at 9 33 45 AM" src="https://github.com/user-attachments/assets/c76b0f63-758f-4609-93f4-38a9e3e95259">

Once changes are committed you also need to **Sync** changes (**_no message needed here_**). 

<img width="871" alt="Screenshot 2024-09-17 at 9 33 53 AM" src="https://github.com/user-attachments/assets/b2bbda58-45a2-469e-b472-cff08455e3fd">

Select **Yes** from the pop up message about pulling/pushing to and from origin/main:

<img width="579" alt="Screenshot 2024-09-17 at 9 34 01 AM" src="https://github.com/user-attachments/assets/a0aa1d9d-9711-4542-b127-3e25955b5f34">

Select **Yes** if prompted about running Git fetch periodically. 

<img width="745" alt="Screenshot 2024-09-17 at 9 34 15 AM" src="https://github.com/user-attachments/assets/0db307e1-f010-4902-9d88-ab52bd29e89f">

Whew! Once you've done all of that you'll see the changes show up in your repository for this assignment in GitHub. I would recommend committing and syncing changes regularly to make sure your work is getting saved!

Again, we'll have a demo video up soon of us doing this last step to submit your work. -NQ

**Comment by eculler** [comment]
@allenmoench I have also put some instructions up on the [wiki](https://github.com/earthlab-education/Earth-Analytics-AY24/wiki/How-To#save-and-submit-your-work). The videos will also go there once I'm done uploading. Thanks @nquarder for the screenshots! Maybe we can write a textbook page for this going forward?

**Comment by eculler** [comment]
@earthlab-education/ea-students A video on how to save your work/sync Codespaces is now posted in [the wiki How To](https://github.com/earthlab-education/Earth-Analytics-AY24/wiki/How-To#save-and-submit-your-work)

**Reply by allenmoench** [reply]
Hi @eculler Are you doing office hours at the moment? I saw the announcement about office hours on Tuesday, but I wasn't sure if that applied to this week. In any case, I'm trying to connect on zoom.

**Reply by eculler** [reply]
Hi @allenmoench sorry about the confusion on this - the Tuesday hours will start next week!

**Reply by allenmoench** [reply]
Thanks for the information! I think I can figure it out now.
